src/main/java/htsjdk/samtools/seekablestream/SeekablePathStream.java

```diff
@@ -0,0 +1,108 @@
+package htsjdk.samtools.seekablestream;
+
+import htsjdk.samtools.util.Log;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+
+public class SeekablePathStream extends SeekableStream {
+
+    private final static Log LOG = Log.getInstance(SeekablePathStream.class);
+
+    /**
+     * Collection of all open instances.  SeekablePathStream objects are usually open and kept open for the
+     * duration of a session.  This collection supports a method to close them all.
+     */
+    static Collection<SeekablePathStream> allInstances =
+            Collections.synchronizedCollection(new HashSet<SeekablePathStream>());
+
+    private final Path path;
+    private final SeekableByteChannel sbc;
+    private final long length;
+    private final ByteBuffer oneByteBuf = ByteBuffer.allocate(1);
+
+    public SeekablePathStream(final Path path) throws IOException {
+        this.path = path;
+        this.sbc = Files.newByteChannel(path);
+        this.length = sbc.size();
+        allInstances.add(this);
+    }
+
+    public long length() {
+        return length;
+    }
+
+    public boolean eof() throws IOException {
+        return length() == position();
+    }
+
+    public void seek(final long position) throws IOException {
+        sbc.position(position);
+    }
+
+    public long position() throws IOException {
+        return sbc.position();
+    }
+
+    @Override
+    public long skip(long n) throws IOException {
+        long initPos = position();
+        sbc.position(initPos + n);
```

Comment:
It will succeed, but then any attempt to read will get an EOF. This is the same as `SeekableFileStream`, and is specified here: https://docs.oracle.com/javase/8/docs/api/java/nio/channels/SeekableByteChannel.html#position-long-